code/model/modelread.cpp

```diff
@@ -4742,21 +4742,26 @@ void model_update_instance(int model_instance_num, int sub_model_num, submodel_i
 	// Set the "blown out" flags
 	if ( flags[Ship::Subsystem_Flags::No_disappear] ) {
 		smi->blown_off = false;
+		sm->blown_off = 0;
```

Comment:
My understanding of the model instance code is pretty limited but I think one of the main points was to avoid changing the actual `polymodel` data for a specific model so this is probably not correct.

Reply:
That seems logical. When I traced this bug back to its first occurrence, it appeared when the model instance code was added, so it appears that this functionality never got ported to using the new system. For my fix I simply added to `model_update_instance()` which was from the `model_set_instance()` function. I would be happy to have a fresh set of eyes on coming up with a better fix, especially since this is a visually glaring bug when ships explode and have all their un-destroyed models pop back into view.